musl libc support

OpenDMARC fails to compile against musl libc. Tested against musl 1.2.2 on Gentoo amd64 with OpenDMARC 1.3.3 (from the repositories) and the latest commit (51808b6f970d802c0f252bab33ab692c217662b8) at the time of writing.

Here's the compiler error for 1.3.3:

```
/bin/sh ../libtool  --tag=CC   --mode=compile x86_64-gentoo-linux-musl-gcc -DHAVE_CONFIG_H -I. -I..  -I. -I.. -DCONFIG_BASE=\"/etc\"  -I /usr/include/spf2 -O2 -pipe -march=native -c -o opendmarc_policy.lo opendmarc_policy.c
libtool: compile:  x86_64-gentoo-linux-musl-gcc -DHAVE_CONFIG_H -I. -I.. -I. -I.. -DCONFIG_BASE=\"/etc\" -I /usr/include/spf2 -O2 -pipe -march=native -c opendmarc_policy.c  -fPIC -DPIC -o .libs/opendmarc_policy.o
opendmarc_policy.c: In function 'opendmarc_policy_query_dmarc_xdomain':
opendmarc_policy.c:660:8: error: 'NETDB_INTERNAL' undeclared (first use in this function)
  660 |   case NETDB_INTERNAL:
      |        ^~~~~~~~~~~~~~
opendmarc_policy.c:660:8: note: each undeclared identifier is reported only once for each function it appears in
opendmarc_policy.c: In function 'opendmarc_policy_query_dmarc':
opendmarc_policy.c:774:8: error: 'NETDB_INTERNAL' undeclared (first use in this function)
  774 |   case NETDB_INTERNAL:
      |        ^~~~~~~~~~~~~~
opendmarc_policy.c: In function 'opendmarc_policy_to_buf':
opendmarc_policy.c:1778:8: error: 'NETDB_INTERNAL' undeclared (first use in this function)
 1778 |   case NETDB_INTERNAL:
      |        ^~~~~~~~~~~~~~
make[3]: *** [Makefile:529: opendmarc_policy.lo] Error 1
```

And when trying to compile the latest commit (51808b6f970d802c0f252bab33ab692c217662b8):

```
libtool: compile:  gcc -Wall -DHAVE_CONFIG_H -I. -I.. -I. -I.. -DCONFIG_BASE=\"/usr/local/etc\" -g -O2 -MT opendmarc_dns.lo -MD -MP -MF .deps/opendmarc_dns.Tpo -c opendmarc_dns.c  -fPIC -DPIC -o .libs/opendmarc_dns.o
opendmarc_dns.c: In function 'dmarc_dns_get_record':
opendmarc_dns.c:172:18: error: 'NETDB_SUCCESS' undeclared (first use in this function); did you mean 'EXIT_SUCCESS'?
  172 |     *reply_ptr = NETDB_SUCCESS;
      |                  ^~~~~~~~~~~~~
      |                  EXIT_SUCCESS
opendmarc_dns.c:172:18: note: each undeclared identifier is reported only once for each function it appears in
make[3]: *** [Makefile:525: opendmarc_dns.lo] Error 1
```

Similar to the [patch for socat][1], here's a patch for OpenDMARC 1.3.3 that makes it compile and work correctly. As of the latest commit it also tries to access NETDB_SUCCESS in opendmarc_dns.c which doesn't exist either (looks like it should be 0?), so this patch isn't enough anymore by itself, but it might be a start. (well, I'm not sure if this is the correct way to fix it in the first place)

```patch
--- a/libopendmarc/opendmarc_internal.h 2020-11-26 16:40:04.789000000 +0100
+++ b/libopendmarc/opendmarc_internal.h 2020-11-26 16:40:09.672709585 +0100
@@ -102,6 +102,10 @@
 # endif 
 # define bool int
 
+#ifndef NETDB_INTERNAL
+#define NETDB_INTERNAL (-1)
+#endif
+
 /*
 ** Beware that some Linux versions incorrectly define 
 ** MAXHOSTNAMELEN as 64, but DNS lookups require a length
```

[1]: https://bugs.gentoo.org/703212#c4

--- a/libopendmarc/opendmarc_internal.h
+++ b/libopendmarc/opendmarc_internal.h
@@ -101,6 +101,15 @@
 #       define MAYBE		(2)
 # endif 
 # define bool int
+
+#ifndef NETDB_INTERNAL
+# define NETDB_INTERNAL (-1)
+#endif
+
+#ifndef NETDB_SUCCESS
+# define NETDB_SUCCESS (0)
+#endif
+
 /*
 ** Beware that some Linux versions incorrectly define 
 ** MAXHOSTNAMELEN as 64, but DNS lookups require a length

